### PR TITLE
Update Developer's Guide for v1.4

### LINKED ISSUE #212: Review DG for v1.4
The DG is somewhat outdated and requires a review.

@lrnshk gotta clarify "case" in glossary

### PR DESCRIPTION
**Pull request by lrnshk**
Fixes #212

### CODE REVIEW

**Comment by wei2912** [comment]
This should be merged before https://github.com/AY2122S1-CS2103-W14-2/tp/pull/223, which will update `ModelClassDiagram.puml`.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -516,48 +534,33 @@ testers are expected to do more *exploratory* testing.
...
+### Adding a new Person
 
-2. Deleting a person while search results are being shown
-   1. Prerequisites: Find specific person(s) using the `find` command.
+1. Adding a person
 
-   2. Test case: `delete 1`<br>
-      Expected: First person displayed in the search results is deleted from the application. Details of the deleted person shown in the status message.
+   1. Test case: `add n/John Doe p/91234567 e/johndoe@gmail.com cn/1 ha/123 Waterloo`<br>
+       Expected: New person is added with the given details. Details of the edited person shown in the
```
Comment (frederickpek) [inline docs/DeveloperGuide.md:542]: Details of the "added" person instead.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -516,48 +534,33 @@ testers are expected to do more *exploratory* testing.
...
+       status message.
 
-   3. Test case: `delete 1 2`<br>
-      Expected: First and second persons displayed in the search results are deleted from the application. Details of the deleted persons shown in the status message.
+   2. Test case: `add n/John Doe p/91234567 e/johndoe@gmail.com cn/3 ha/123 Waterloo wa/456 Waterloo`<br>
+       Expected: New person is added with the given details. Details of the edited person shown in the
```
Comment (frederickpek) [inline docs/DeveloperGuide.md:546]: Details of the "added" person instead.

**Review by frederickpek — CHANGES_REQUESTED** [review]
Thanks for making the DG collectively easy to read, especially the usage of Enforcement mode! 

Just some minor considerations before merging.

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -58,19 +59,27 @@ Person *--> "0..1" ShnPeriod
 Person *--> "0..1" NextOfKinName
 Person *--> "0..1" NextOfKinPhone
 Person *--> "0..1" NextOfKinAddress
```
[the remark this answers is not in the document]
Comment (frederickpek) [reply docs/diagrams/ModelClassDiagram.puml:61]: @luffingluffy Has updated this in his refactoring PR.

**Review by frederickpek — APPROVED** [review]
Looks good to merge on my end!

**Review by jennibearduit — APPROVED** [review]
LGTM!

**Pull request merged**